application/helpers/common_helper.php

```diff
@@ -2335,9 +2335,8 @@ function languageDropdown($surveyid, $selected)
     $baselang = Survey::model()->findByPk($surveyid)->language;
     array_unshift($slangs, $baselang);
     $html = "<select class='listboxquestions' name='langselect' onchange=\"window.open(this.options[this.selectedIndex].value, '_top')\">\n";
-
     foreach ($slangs as $lang) {
-        $link = Yii::app()->homeUrl . ("/admin/dataentry/sa/view/surveyid/" . $surveyid . "/lang/" . $lang);
+        $link = Yii::app()->createUrl("admin/dataentry/sa/view/surveyid/". $surveyid . "/lang/" . $lang);
```

Comment:
I personnaly prefer `Yii::app()->createUrl("admin/dataentry/sa/view/surveyid/". $surveyid, ['lang' => $lang])` for me lang is a parameter, i know sureyid too, but less clear

Reply:
These are written with different patterns all over the codebase.... We should set a std

Reply:
Yes , i know …

Reply:
Something for the code quality guide?

Reply:
Defintely, I will let you set the standard :)